File: BOJ/boj14719.cpp
/*
 * 14719 빗물
 */
#include <iostream>
using namespace std;
int h,w,res = 0;
int board[501];
void solve() {
	cin >> h >> w;
	for (int i = 0; i < w; i++) {
		cin >> board[i];
	}

	for (int i = 1; i < w; i++) {
		int left = i;
		int right = i;
		int lmx=board[i], rmx=board[i];
		//왼쪽 최댓값
		while (left >= 0) {
			lmx = max(lmx,board[left--]);
		}
		//오른쪽 최댓값
		while (right < w) {
			rmx = max(rmx,board[right++]);
		}
		res += (min(lmx,rmx) - board[i]);
	}
	cout << res << "\n";
}
int main(void){
	ios::sync_with_stdio(false);
    cin.tie(nullptr);
	solve();
	return 0;
}

File: BOJ/boj14719.java
/*
 * boj 14719 빗물
 */
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class boj14719 {
    public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		//input
		String[] info = br.readLine().split(" ");
		String[] boardInfo = br.readLine().split(" ");
		int w = Integer.parseInt(info[1]);
		int[] board = new int[w+1];
		for (int i = 0; i < w; i++) {
			board[i] = Integer.parseInt(boardInfo[i]);
		}

		//search
		int res = 0;
		for (int i = 1; i < w; i++) {
			int left = i, right = i;
			int lmx = board[i], rmx = board[i];
			while(left >= 0) {
				lmx = Math.max(lmx, board[left--]);
			}
			while(right < w) {
				rmx = Math.max(rmx, board[right++]);
			}
			res += (Math.min(lmx,rmx) - board[i]);
		}
		System.out.println(res);
		br.close();
	}
}
